Crash report 3.9-rc-1: NPE in MediaBrowserActivity

```
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x4000010 (has extras) } in org.wordpress.android.ui.media.MediaBrowserActivity$1@424ab198
       at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:778)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5086)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.ui.media.MediaBrowserActivity$1.onReceive(MediaBrowserActivity.java:80)
       at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:768)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5086)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/WordPress/src/main/java/org/wordpress/android/WordPressDB.java
+++ b/WordPress/src/main/java/org/wordpress/android/WordPressDB.java
@@ -1649,6 +1649,10 @@ public class WordPressDB {
                 new String[]{"delete", blogId});
     }
 
+    public boolean hasMediaDeleteQueueItems(int blogId) {
+        return SqlUtils.boolForQuery(db, "SELECT 1 FROM " + MEDIA_TABLE + " WHERE uploadState=? AND blogId=?",
+                new String[]{"delete", Integer.toString(blogId)});
+    }
 
     public int getWPCOMBlogID() {
         int id = -1;

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
@@ -32,6 +32,7 @@ import android.widget.Toast;
 import org.wordpress.android.Constants;
 import org.wordpress.android.R;
 import org.wordpress.android.WordPress;
+import org.wordpress.android.models.Blog;
 import org.wordpress.android.models.FeatureSet;
 import org.wordpress.android.ui.WPDrawerActivity;
 import org.wordpress.android.ui.media.MediaAddFragment.MediaAddFragmentCallback;
@@ -55,8 +56,8 @@ import java.util.Set;
  * Accessible via the menu drawer as "Media"
  */
 public class MediaBrowserActivity extends WPDrawerActivity implements MediaGridListener,
-        MediaItemFragmentCallback, OnQueryTextListener, OnActionExpandListener, MediaEditFragmentCallback,
-        MediaAddFragmentCallback {
+        MediaItemFragmentCallback, OnQueryTextListener, OnActionExpandListener,
+        MediaEditFragmentCallback, MediaAddFragmentCallback {
     private static final String SAVED_QUERY = "SAVED_QUERY";
 
     private MediaGridFragment mMediaGridFragment;
@@ -74,17 +75,19 @@ public class MediaBrowserActivity extends WPDrawerActivity implements MediaGridL
     private BroadcastReceiver mReceiver = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
-            String action = intent.getAction();
-            if (ConnectivityManager.CONNECTIVITY_ACTION.equals(action)) {
+            if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
                 // Coming from zero connection. Continue what's pending for delete
-                String blogId = String.valueOf(WordPress.getCurrentBlog().getLocalTableBlogId());
-                if (WordPress.wpDB.getMediaDeleteQueueItems(blogId).getCount() > 0) {
-                    startMediaDeleteService();
+                Blog currentBlog = WordPress.getCurrentBlog();
+
+                if (currentBlog != null) {
+                    String blogId = String.valueOf(currentBlog.getLocalTableBlogId());
+                    if (WordPress.wpDB.getMediaDeleteQueueItems(blogId).getCount() > 0) {
+                        startMediaDeleteService();
+                    }
                 }
             }
         }
     };
-
 
     @Override
     public void onCreate(Bundle savedInstanceState) {

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
@@ -77,13 +77,9 @@ public class MediaBrowserActivity extends WPDrawerActivity implements MediaGridL
         public void onReceive(Context context, Intent intent) {
             if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
                 // Coming from zero connection. Continue what's pending for delete
-                Blog currentBlog = WordPress.getCurrentBlog();
-
-                if (currentBlog != null) {
-                    String blogId = String.valueOf(currentBlog.getLocalTableBlogId());
-                    if (WordPress.wpDB.getMediaDeleteQueueItems(blogId).getCount() > 0) {
-                        startMediaDeleteService();
-                    }
+                int blogId = WordPress.getCurrentLocalTableBlogId();
+                if (blogId != -1 && WordPress.wpDB.hasMediaDeleteQueueItems(blogId)) {
+                    startMediaDeleteService();
                 }
             }
         }

